airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/ComponentRunner.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.components
+
+import java.time.Duration
+import java.util.concurrent.atomic.AtomicReference
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * [ComponentRunner] orchestrates [ProducerComponent] and [ConsumerComponent] instances to collect
+ * records and states.
+ *
+ * Constructor args:
+ * - [name] is decorative-only and is used to name the threads spawned by this object;
+ * - [producerBuilder] is used to instantiate [ProducerComponent];
+ * - [consumerBuilder] is used to instantiate [ConsumerComponent];
+ * - [maxTime] is the global timeout for a [collect] method call;
+ * - [comparator] is used by [collectRepeatedly] to compare relative progress between two states.
+ */
+class ComponentRunner<R, S>(
```

Comment:
Can we rename this? Perhaps RecordOrchestrator/RecordFetcher or something to indicate that it is fetching records. 

Same with ProducerComponent/ConsumerComponent